frontend/src/utils/account-send.ts

```diff
@@ -150,7 +151,7 @@ export const storeSend = async ({
   const count =
     ((await localforage.getItem(
       `${LOCALFORAGE_ACCOUNT_SEND_KEY_PREFIX}-count-${senderAddress}-${chainId}`
-    )) as number) || 0;
+    )) as string) || new RandomNumber().value.toString();
```

Comment:
Would it be even more secure if we stored the random number for each entry? There is no requirement we have to increment by 1.

Reply:
Good question, but I don't think so, because in either case an attacker with access to local storage knows all the count values (they are either all in local storage if we use random each time, or they can all be derived)

Reply:
True, this is a little out of my wheelhouse, but is incrementing a random number by 1 less random than picking x random numbers? And if it is less random is an attacker more likely to observe two encryption counts that are the same.

Reply:
>  is incrementing a random number by 1 less random than picking x random numbers? 

Yea, by definition 😛 since incrementing is of course not random

> And if it is less random is an attacker more likely to observe two encryption counts that are the same.

Observe them *where* though? To observe the encryption counts you need access to local storage. AFAIK there is no granular permissioning here of what e.g. extensions can access. So if an attack can see your local storage:
1. Incrementing count:  They'll see the length of the stored sends array, the initial count that was used, and be able to derive the rest of the encryption counts needed
2. Random count:  They'll see the length of the stored sends array, the *every* count that was used, so there is nothing to derive (the assumption here is that we have to store all random counts in local storage)

Reply:
Ah I think I misread, you said:

> more likely to observe two encryption counts that are the same.

To clarify, we should never have to encryption counts that are the same, because this would result in key reuse

Reply:
Yes exactly they should never be the same. We are relying on the randomness of the initial number count to prevent seeing the same number twice. Right? 

To restate my question in a different way. Would it be less likely to see the same number twice if we generated a new random number for each entry? The answer seems to be yes. And does this greater randomness provide any additional security?

It definitely seems like an edge case to me so it may not matter, but if a user refreshes their localstorage a bunch of times with a good number of sends maybe it is more likely we would see the same count twice.

Thought, I would point this out because it is a pretty small change to generate a new random number for each entry rather than increment.

Reply:
Ah ok I think I see what you're saying. Just to back up for a bit and make sure we're on the same page / have the same context: The security issue arises if we use the same encryption key twice. Our encryption key `k` can be represented as `k = h(v, c)`, i.e. the encryption key is the output of the hash function of two things: a viewing private key `v` and an arbitrary number `c` that we call the encryption count. 

`v` is constant, meaning we MUST rely on unique `c` to avoid getting a duplicate `k`. 

Now let's say we choose `c` with the following algorithm, call it algorithm 1:
- If no previous `c` is found in local storage, use `c = 0`
- If a previous `c` is found in local storage use `c = c_previous + 1`

This works fine. UNLESS local storage is cleared, now we use `c = 0` a second time. This brings us to the modified version of the algorithm, call it algorithm 2:
- If no previous `c` is found in local storage, set `c` to a random 32 byte number.
- If a previous `c` is found in local storage use `c = c + 1`

This means we only use a duplicate key if local storage is cleared AND we somehow generate the same random number next time. And with a 32 byte random number, the odds of that are effectively 0 (it's like guessing a private key).

Now we could instead use algorithm 3:
- Always set `c` to a random 32 byte number.

You might be able to argue that algorithm 3 is a tiny amount more secure than 2 because all `c` values are uniformly distributed, where as the `c` values in 2 are clustered. But whether or not they are clustered doesn't matter, it just matters that they are unique. Consider:
- if you generate `c_0` and increment all the way to `c0 + 1trillion`, then generate a new `c_0`
- The odds of your new `c_0` being in the range of `(c_0_original, c_0_original + 1trillion)` are about 10e-66, which is effectively 0.
- In either case (algo 2 or 3) you have 1 trillion numbers you don't want to avoid a collision with when generating the next `c_0`, so the odds of duplicating a `c` value feel the same in both cases. (Assuming a safe random number generator. Maybe if you can somehow bias it in some direction this argument is weaker, but I'd argue we can trust the random number generator since it's used at the core of umbra too)

Therefore, we should implement whichever approach is simpler and more practical. One upside of the counter is that it uses less local storage, but in practice that is unlikely to matter, so I'll defer to you on whichever you prefer / has a cleaner implementation

Reply:
Thanks, great response! Let's keep it as incrementing